launcher/tests/test_updater.py

```diff
@@ -840,3 +833,90 @@ def test_last_required_reboot_performed_not_required(
     result = updater.last_required_reboot_performed()
     assert result is True
     assert not mocked_error.called
+
+
+@pytest.mark.parametrize(
+    "status, expected, rebooted",
+    [
+        (UpdateStatus.UPDATES_OK, True, True),
+        (UpdateStatus.UPDATES_REQUIRED, True, True),
+        (UpdateStatus.REBOOT_REQUIRED, True, True),
+        (UpdateStatus.UPDATES_FAILED, True, True),
+        (UpdateStatus.UPDATES_OK, True, False),
+        (UpdateStatus.UPDATES_REQUIRED, True, False),
+        (UpdateStatus.REBOOT_REQUIRED, True, False),
+        (UpdateStatus.UPDATES_FAILED, True, False),
+    ],
+)
+@mock.patch("Updater._write_updates_status_flag_to_disk")
+def test_should_run_updater_status_interval_expired(
```

Comment:
This and `test_should_run_updater_status_interval_not_expired` are your most complex tests, and I think they would benefit from a few code comments/test descriptions to explain the test behavior. That said, after starting at them for 15 minutes or so, the behavior seems reasonable to me.

My personal intuition would be to put the `expected` value rightmost in each list, to make the lists a bit more readable (kind of like a truth table), but I don't know if there's a convention you're following here that goes another way.

Reply:
:heavy_check_mark: